src/features/todos.ts

```diff
@@ -1,8 +1,12 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
 
 export const todosSlice = createSlice({
   name: 'todos',
   initialState: [] as Todo[],
-  reducers: {},
+  reducers: {
+    setTodos: (_, { payload }: PayloadAction<Todo[]>) => [...payload],
```

Comment:
The `setTodos` reducer is correctly defined to replace the state with the payload. Ensure that the `Todo` type includes all necessary properties like `id`, `title`, `completed`, and `userId` to match the expected data shape.